install/cluster-operator/047-Crd-kafkaconnector.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: apiextensions.k8s.io/v1alpha1
+kind: CustomResourceDefinition
+metadata:
+  name: kafkaconnectors.kafka.strimzi.io
+  labels:
+    app: strimzi
+spec:
+  group: kafka.strimzi.io
+  versions:
+  - name: v1alpha1
+    served: true
+    storage: false
+  version: v1alpha1
+  scope: Namespaced
+  names:
+    kind: KafkaConnector
+    listKind: KafkaConnectorList
+    singular: kafkaconnector
+    plural: kafkaconnectors
+    shortNames:
+    - kctr
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          type: object
+          properties:
+            connectCluster:
+              type: object
+              properties:
+                url:
+                  type: string
+            class:
+              type: string
+            tasksMax:
+              type: integer
+              minimum: 1
+            topics:
```

Comment:
Just to be sure, this is optional, correct (no topic needed for source connectors)?

Reply:
If we're sticking to the plan that we're having a single CRD for both sources and sinks then yes, it would have to be. Having separate CRD would:

* Allow you to get/watch etc them separately from the kube api server. TBH I don't really see that being advantageous. I would expect that a user is as likely to want to list all the connectors e.g. using `kubectl` as they are to just list sources or sinks.
* Allow different validation schemas for each. But since the only difference would be `topics` that seems to have very little value.

These don't seem like compelling benefits to me.

Reply:
Yes, +1 for one unified CRD type. The REST API also doesn't distinguish between sources and sinks.

Reply:
why "no topic needed for source connectors" if a source connector gets data from an external system and writes to a topic?
Anyway it's something I don't have to teach to you @gunnarmorling :P ... so there is for sure a point I am missing

Reply:
Yes, I just wanted to make sure that `topics` is optional for the exact reason you point out. We're on the same page :)